### PR TITLE
Setup RTC Web Socket Server

### PR DESCRIPTION
**Pull request by 1solomonwakhungu**
*What?*
- Add the ability to connect to the web socket server
- Add various possible requests to the server.
- Validate requests to the server.

*Why?*
- The web socket server allows other services to communicate with the RTC service.
- There will be a variety of requests made to the server and there are different ways of handling each request.
- Request validation is important to ensure the passed data is not corrupted.

*Testing*
- Tested with invalid JSON request
  - Received an error with the exact issue that was wrong with the request
- Tested with invalid request types
  - Received an error stating there was an invalid request type.
- Tested with valid JSON
  - Received an ok message with the appropriate request type.

*Notes*
- Testing was done by sending data to `ws://localhost:8080/ws`
- Will update the docker ports in the future PRs
- Will update unfinished methods in future PRs.

### CODE REVIEW

File: rtc-service/requests/chat-message-request.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package requests
+
+// Request for a user to send a message to a room.
+type ChatMessageRequest struct {
+	UserID  string `json:"userID"`
```
Comment (DedsecKnight) [inline rtc-service/requests/chat-message-request.go]: I believe this should be `UserHandle`

File: rtc-service/requests/join-room-request.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package requests
+
+// Request for a user to join a room.
+type JoinRoomRequest struct {
+	UserID string `json:"userID"`
```
Comment (DedsecKnight) [inline rtc-service/requests/join-room-request.go]: I believe this should be `UserHandle`

File: rtc-service/requests/leave-room-request.go [hunk 3]
```diff
@@ -0,0 +1,24 @@
+package requests
+
+// Struct for the leave-room request.
+// Request for a user to leave a room.
+type LeaveRoomRequest struct {
+	UserID string `json:"userID"`
```
Comment (DedsecKnight) [inline rtc-service/requests/leave-room-request.go]: I believe this should be `UserHandle`

**Review by DedsecKnight — APPROVED** [review]
Other than some of the small stuff, this PR lgtm!. Nice code quality btw :+1:

Also, please resolve merge conflict before merging.

**Review by spierceVR — APPROVED** [review]
Looks great! Seems like you've done this before lol

**Pull request merged**